src/libAtomVM/externalterm.c

```diff
@@ -421,6 +422,47 @@ static int serialize_term(uint8_t *buf, term t, GlobalContext *glb)
             WRITE_32_UNALIGNED(buf + k + 8, term_get_external_node_creation(t));
         }
         return k + 12;
+    } else if (term_is_local_reference(t)) {
+        if (!IS_NULL_PTR(buf)) {
+            buf[0] = NEWER_REFERENCE_EXT;
+        }
+        size_t k = 1;
+        if (!IS_NULL_PTR(buf)) {
+            WRITE_16_UNALIGNED(buf + k, 2); // len
```

Comment:
Question: is it more readable writing ?
```c
uint16_t len = 2;
WRITE_16_UNALIGNED(buf + k, len);
```

Reply:
Not sure if it's more readable but there was a bug below related to this (len wasn't encoded). I adapted tests accordingly.